File: package-admin/src/pt_pump_up_admin/crud/Result.py
from pt_pump_up_admin.crud.CRUD import CRUD
from pt_pump_up_admin.dataset import Dataset


class Result(CRUD):
    def __init__(self,
                 id: int = None,
                 metric: str = None,
                 value: float = None,
                 train_dataset: Dataset = None,
                 validation_dataset: Dataset = None,
                 test_dataset: Dataset = None,
                 ) -> None:

        super().__init__("result",
                         id=id,
                         metric=metric,
                         value=value,
                         train_dataset_id=train_dataset.id if train_dataset else None,
                         validation_dataset_id=validation_dataset.id if validation_dataset else None,
                         test_dataset_id=test_dataset.id if test_dataset else None)

File: web/app/Models/Result.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Result extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'metric',
        'value',
        'machine_learning_model_id',
        'train_dataset_id',
        'validation_dataset_id',
        'test_dataset_id',
    ];

    /**
     * The attributes that should be cast to native types.
     *
     * @var array
     */
    protected $casts = [
        'id' => 'integer',
        'value' => 'float',
        'machine_learning_model_id' => 'integer',
        'train_dataset_id' => 'integer',
        'validation_dataset_id' => 'integer',
        'test_dataset_id' => 'integer',
    ];

    public function machineLearningModel(): BelongsTo
    {
        return $this->belongsTo(MachineLearningModel::class);
    }

    public function trainDataset(): BelongsTo
    {
        return $this->belongsTo(Dataset::class);
    }

    public function validationDataset(): BelongsTo
    {
        return $this->belongsTo(Dataset::class);
    }

    public function testDataset(): BelongsTo
    {
        return $this->belongsTo(Dataset::class);
    }
}
